packages/flutter/lib/src/painting/borders.dart

```diff
@@ -75,8 +76,33 @@ class BorderSide {
   const BorderSide({
     this.color: const Color(0xFF000000),
     this.width: 1.0,
-    this.style: BorderStyle.solid
-  });
+    this.style: BorderStyle.solid,
+  }) : assert(color != null),
+       assert(width != null),
+       assert(width >= 0.0),
+       assert(style != null);
+
+  /// Creates a [BorderSide] that represents the addition of the two given
+  /// [BorderSide]s.
+  ///
+  /// It is only valid to call this if [canMerge] returns true for the two
+  /// sides.
+  ///
+  /// If both sides are null, then this will return null.
+  static BorderSide merge(BorderSide a, BorderSide b) {
```

Comment:
Document the behavior of merging a non-null border with null.

Reply:
done